packages/@coorpacademy-app-review/src/actions/ui/answers.ts

```diff
@@ -1,16 +1,62 @@
 import {Dispatch} from 'redux';
+import {flatten, get, includes, pull} from 'lodash/fp';
+import type {StoreState} from '../../reducers';
 
-export const EDIT_BASIC = '@@answer/EDIT_BASIC' as const;
-export type EditBasicAction = {
-  type: typeof EDIT_BASIC;
+const EDIT_QCM = '@@answer/EDIT_QCM' as const;
+const EDIT_QCM_GRAPHIC = '@@answer/EDIT_QCM_GRAPHIC' as const;
+const EDIT_QCM_DRAG = '@@answer/EDIT_QCM_DRAG' as const;
+const EDIT_TEMPLATE = '@@answer/EDIT_TEMPLATE' as const;
+const EDIT_BASIC = '@@answer/EDIT_BASIC' as const;
+const EDIT_SLIDER = '@@answer/EDIT_SLIDER' as const;
+
+export const ANSWER_EDIT = {
+  qcm: EDIT_QCM,
+  qcmGraphic: EDIT_QCM_GRAPHIC,
+  qcmDrag: EDIT_QCM_DRAG,
+  template: EDIT_TEMPLATE,
+  basic: EDIT_BASIC,
+  slider: EDIT_SLIDER
+} as const;
+
+export type EditAnswerAction = {
+  type:
+    | typeof EDIT_QCM
+    | typeof EDIT_QCM_GRAPHIC
+    | typeof EDIT_QCM_DRAG
+    | typeof EDIT_TEMPLATE
+    | typeof EDIT_BASIC
+    | typeof EDIT_SLIDER;
   payload: string[];
 };
 
+const buildAnswer = (userAnswers: string[], questionType: string, newValue: string): string[] => {
+  switch (questionType) {
+    case 'qcm':
+    case 'qcmGraphic':
+    case 'qcmDrag': {
+      return includes(newValue, userAnswers)
+        ? pull(newValue, userAnswers)
+        : flatten([...userAnswers, newValue]);
+    }
+    case 'basic':
+      return [newValue];
+    default:
+      return [];
```

Comment:
https://github.com/CoorpAcademy/components/blob/master/packages/%40coorpacademy-player-store/src/actions/ui/answers.js#L37-L67

```suggestion
    case 'basic':
    case 'slider':
      return [newValue];
    case 'template':
      return newValue;
    default:
      return [];
```

Reply:
@esa-coorp la PR ne traite pas du slider